src/libraries/Microsoft.Extensions.Logging/gen/LoggerMessageGenerator.Emitter.cs

```diff
@@ -443,63 +445,56 @@ static string GetLogLevel(LoggerMethod lm)
                 }
             }
 
-            private void GenEnumerationHelper(LoggerClass lc)
+            private void GenEnumerationHelper()
             {
-                foreach (LoggerMethod lm in lc.Methods)
+                if (_needEnumerationHelper)
                 {
-                    if (UseLoggerMessageDefine(lm))
-                    {
-                        foreach (LoggerParameter p in lm.TemplateParameters)
-                        {
-                            if (p.IsEnumerable)
-                            {
                                 _builder.Append($@"
-        [{_generatedCodeAttribute}]
-        private static string __Enumerate(global::System.Collections.IEnumerable? enumerable)
+[{_generatedCodeAttribute}]
+internal static class __LoggerMessageGenerator
```

Comment:
Just thinking out loud here, but I almost wonder if this deserves to be a public API in our library instead of generating this code into the user's project.

I started by thinking "is putting this in the global namespace the right thing to do?" And then that quickly led to "why are we generating this code when it never changes based on the user's types?"

Reply:
In the very first API review proposal (Refer to comment https://github.com/dotnet/runtime/issues/36022#issuecomment-796888508), we proposed this static `ArgumentFormatter.Enumerate(..)` helper method which we said would only be used in the generated code.

The consensus back then was for such cases to just generate if/when needed.